transport/cluster.go

```diff
@@ -0,0 +1,288 @@
+package transport
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"strconv"
+	"time"
+
+	"scylla-go-driver/frame"
+	. "scylla-go-driver/frame/response"
+
+	"go.uber.org/atomic"
+)
+
+type PeerMap = map[string]*Node
+
+type refreshHandler chan struct{}
+
+type Cluster struct {
+	peers         atomic.Value // PeerMap
+	control       *Conn
+	cfg           ConnConfig
+	events        responseHandler // TODO: should events have their own type of chan or is responseHandler fine?
+	handledEvents []frame.EventType
+	knownHosts    []string // TODO: do we want to update known hosts with node info from cluster?
```

Comment:
I understand this is the list of host for control connection. Let's keep this code simple as it already is quite a big pr.
We can definitely add this kind of feature later.